pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/ShadowReplicator.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.broker.service.persistent;
+
+
+import io.netty.buffer.ByteBuf;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.bookkeeper.mledger.AsyncCallbacks;
+import org.apache.bookkeeper.mledger.Entry;
+import org.apache.bookkeeper.mledger.ManagedCursor;
+import org.apache.bookkeeper.mledger.ManagedLedgerException;
+import org.apache.bookkeeper.mledger.impl.PositionImpl;
+import org.apache.pulsar.broker.PulsarServerException;
+import org.apache.pulsar.broker.service.BrokerService;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.impl.MessageIdImpl;
+import org.apache.pulsar.client.impl.MessageImpl;
+import org.apache.pulsar.client.impl.PulsarClientImpl;
+import org.apache.pulsar.common.util.Codec;
+
+/**
+ *  Replicate messages to shadow topic.
+ */
+@Slf4j
+public class ShadowReplicator extends PersistentReplicator {
+
+    public ShadowReplicator(String shadowTopic, PersistentTopic sourceTopic, ManagedCursor cursor,
+                            BrokerService brokerService, PulsarClientImpl replicationClient)
+            throws PulsarServerException {
+        super(brokerService.pulsar().getConfiguration().getClusterName(), sourceTopic, cursor,
+                brokerService.pulsar().getConfiguration().getClusterName(), shadowTopic, brokerService,
+                replicationClient);
+    }
+
+    /**
+     * @return Producer name format : replicatorPrefix-localTopic-->remoteTopic
+     */
+    @Override
+    protected String getProducerName() {
+        return replicatorPrefix + "-" + localTopicName + REPL_PRODUCER_NAME_DELIMITER + remoteTopicName;
+    }
+
+    @Override
+    protected void readEntries(Producer<byte[]> producer) {
+        // Shadow replicator should skip all backlog
+        this.cursor.asyncResetCursor(PositionImpl.LATEST, false, new AsyncCallbacks.ResetCursorCallback() {
```

Comment:
If the shadow topic can skip all the backlogs. Can we use the non-durable cursor?